src/main/scala/akka/persistence/cassandra/query/journal/EventsByPersistenceIdPublisher.scala

```diff
@@ -0,0 +1,201 @@
+
+import akka.actor.Props
+import akka.persistence.PersistentRepr
+import akka.persistence.query.EventEnvelope
+import akka.serialization.SerializationExtension
+import com.datastax.driver.core.utils.Bytes
+import com.datastax.driver.core.{ResultSet, Row}
+
+private[journal] object EventsByPersistenceIdPublisher {
+  def props(
+      persistenceId: String,
+      fromSeqNr: Long,
+      toSeqNr: Long,
+      refreshInterval: Option[FiniteDuration],
+      maxBufSize: Long,
+      config: CassandraReadJournalConfig): Props =
+    Props(new EventsByPersistenceIdPublisher(persistenceId, fromSeqNr, toSeqNr, refreshInterval, maxBufSize, config))
+}
+
+// TODO: Decouple database.
+// TODO: Query index tables instead.
+// TODO: Generic message iterator to handle different tables etc.
+private[journal] class EventsByPersistenceIdPublisher(
+    persistenceId: String,
+    fromSeqNr: Long,
+    toSeqNr: Long,
+    refreshInterval: Option[FiniteDuration],
+    maxBufSize: Long,
+    override val config: CassandraReadJournalConfig)
+  extends QueryActorPublisher[EventEnvelope, Long](refreshInterval, maxBufSize)
+  with CassandraReadStatements {
+
+  import config._
+
+  private[this] val cluster = clusterBuilder.build
+  private[this] val session = cluster.connect()
+  private[this] val serialization = SerializationExtension(context.system)
+
+  private[this] val preparedSelectMessages = session.prepare(selectMessages).setConsistencyLevel(readConsistency)
+  private[this] val preparedSelectDeletedTo = session.prepare(selectDeletedTo).setConsistencyLevel(readConsistency)
+  private[this] val preparedCheckInUse = session.prepare(selectInUse).setConsistencyLevel(readConsistency)
+
+  private[this] val step = 50l
+
+  override protected def query(state: Long, max: Long): Future[Vector[EventEnvelope]] = {
+    implicit val ec = context.dispatcher
+
+    // TODO: Async?
+    Future {
+      val from = state
+      val to = Math.min(Math.min(state + step, toSeqNr), state + max)
+      val ret = (state to to)
+        .zip(new MessageIterator(persistenceId, from, to, maxBufSize).toVector)
+        .map(r => toEventEnvelope(r._2, r._1 - 1))
+        .toVector
+
+      ret
+    }
+  }
+
+  override protected def initialState: Long = Math.max(1, fromSeqNr)
+
+  override def updateBuffer(
+      buffer: Vector[EventEnvelope],
+      newBuffer: Vector[EventEnvelope],
+      state: Long): (Vector[EventEnvelope], Long) = {
+    val newState = if (newBuffer.isEmpty) state else newBuffer.last.sequenceNr + 1
+    (buffer ++ newBuffer, newState)
+  }
+
+  override protected def completionCondition(state: Long): Boolean = state > toSeqNr
+
+  private[this] def toEventEnvelope(persistentRepr: PersistentRepr, offset: Long): EventEnvelope =
+    EventEnvelope(offset, persistentRepr.persistenceId, persistentRepr.sequenceNr, persistentRepr.payload)
+
+
+
+
+
+
+
+  // TODO: THE BELOW WAS BLINDLY COPIED FROM CASSANDRAJOURNAL AND RECOVERY
```

Comment:
This is a temporary implementation. We will need to refactor the queries. This PR does not focus on those aspects so therefore the temporary code below is introduced. It will be removed when index tables and the real query capabilities are added.